src/runner/fixture-hook-controller.ts

```diff
@@ -59,30 +59,53 @@ export default class FixtureHookController {
         return !!item && item.runningFixtureBeforeHook;
     }
 
+    private async _runFixtureBeforeHook (item: FixtureState, fn: Function): Promise<boolean> {
+        if (!fn)
+            return true;
+
+        item.runningFixtureBeforeHook = true;
+
+        try {
+            await fn(item.fixtureCtx);
+        }
+        catch (err) {
+            item.fixtureBeforeHookErr = processTestFnError(err);
+        }
+
+        item.runningFixtureBeforeHook = false;
+
+        return !item.fixtureBeforeHookErr;
+    }
+
+    private async _runFixtureAfterHook (item: FixtureState, fn: Function | null, testRun: TestRun): Promise<void> {
+        if (!fn)
+            return;
+
+        testRun.phase = TEST_RUN_PHASE.inFixtureAfterHook;
+
+        try {
+            await fn(item.fixtureCtx);
+        }
+        catch (err) {
+            testRun.addError(processTestFnError(err));
+        }
+    }
+
     public async runFixtureBeforeHookIfNecessary (testRun: TestRun): Promise<boolean> {
         const fixture = testRun.test.fixture;
         const item    = this._getFixtureMapItem(testRun.test);
 
         if (item) {
-            const shouldRunBeforeHook = !item.started && fixture.beforeFn;
+            const shouldRunBeforeHook = !item.started;
 
             item.started = true;
 
-            if (shouldRunBeforeHook) {
-                item.runningFixtureBeforeHook = true;
-
-                try {
-                    await (fixture.beforeFn as Function)(item.fixtureCtx);
-                }
-                catch (err) {
-                    item.fixtureBeforeHookErr = processTestFnError(err);
-                }
-
-                item.runningFixtureBeforeHook = false;
-            }
+            const success = shouldRunBeforeHook
+                            && await this._runFixtureBeforeHook(item, fixture.globalBeforeFn as Function)
```

Comment:
The fixture's before/after functions should be executed only one time. The fixture's hooks overrides global ones.

Reply:
It was decided that the hooks would be executed independently of each other.